Cabal-syntax/src/Distribution/Types/VersionInterval/Legacy.hs

```diff
@@ -1,9 +1,16 @@
 {-# LANGUAGE DeriveDataTypeable #-}
--- | This is old version of "Distribution.Types.VersionInterval" module.
+
+-- | This module implements a view of a 'VersionRange' as a finite
+-- list of separated version intervals and provides the Boolean
+-- algebra operations union, intersection, and complement.
 --
--- It will be removed in @Cabal-3.8@.
+-- It interprets the caret operator @^>=x.y@ as simply @==x.y.*@.
+-- Until @Cabal < 3.6@, this module was called "Distribution.Types.VersionInterval".
+-- The current module "Distribution.Types.VersionInterval" (refurbished since
+-- @Cabal >= 3.6@) makes some effort to preserve the caret operator,
+-- but so far does not expose the Boolean algebra structure.
 --
-module Distribution.Types.VersionInterval.Legacy {-# DEPRECATED "Use Distribution.Types.VersionInterval instead" #-} (
```

Comment:
Yeah, we had also discussions on whether to rename this module, during a jitsi meeting.  The outcome was to do the minimal changes vs. 3.6.

If you are interested in this whole complex, I recommend reading the discussion at #7916.

> Out of curiosity, there is some (urgent?) reason to include this in the new release?

Yes, because otherwise the released 3.8 says that this module "will be removed in 3.8", seemingly self-contradictory.